tests/unit/base/trained_model_serialization_unittest.cc.jinja2

```diff
@@ -0,0 +1,210 @@
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/io/SerializableAsciiFile.h>
+#include <shogun/base/class_list.h>
+#include "utils/Utils.h"
+
+#include <shogun/classifier/AveragedPerceptron.h>
+#include <shogun/classifier/LDA.h>
+#include <shogun/classifier/svm/LibLinear.h>
+#include <shogun/classifier/LPBoost.h>
+#include <shogun/classifier/LPM.h>
+#include <shogun/classifier/svm/NewtonSVM.h>
+#include <shogun/classifier/Perceptron.h>
+#include <shogun/classifier/svm/SGDQN.h>
+#include <shogun/classifier/svm/SVMLin.h>
+#include <shogun/classifier/svm/SVMOcas.h>
+#include <shogun/classifier/svm/SVMSGD.h>
+#include <shogun/classifier/LPBoost.h>
+#include <shogun/classifier/LPM.h>
+#include <shogun/regression/svr/LibLinearRegression.h>
+#include <shogun/regression/LinearRidgeRegression.h>
+
+using namespace shogun;
+
+class TrainedModelSerializationTest : public testing::Test
+{
+	protected:
+		static void SetUpTestCase()
+		{
+			/* generate train and test data */
+			CMath::init_random(5);
+
+			/* binary classification data */
+			SGMatrix<float64_t> X_bin(2, N_TRAIN);
+			SGMatrix<float64_t> X_test_bin(2, N_TEST);
+			SGVector<float64_t> Y_bin(N_TRAIN);
+
+			for (index_t i=0; i<N_TRAIN/2; ++i) {
+				X_bin(0, 2*i)=CMath::randn_double()*0.2;
+				X_bin(1, 2*i)=CMath::randn_double()*0.2;
+				Y_bin[2*i]=1.0;
+
+				X_bin(0, 2*i+1)=1.0+CMath::randn_double()*0.2;
+				X_bin(1, 2*i+1)=2.0+CMath::randn_double()*0.2;
+				Y_bin[2*i+1]=-1.0;
+			}
+
+			for (index_t i=0; i<N_TEST; ++i) {
+				X_test_bin(0, i)=CMath::random(-1.0, 2.0);
+				X_test_bin(1, i)=CMath::random(-1.0, 3.0);
+			}
+
+			feat_train_bin=new CDenseFeatures<float64_t>(X_bin);
+			feat_test_bin=new CDenseFeatures<float64_t>(X_test_bin);
+			label_train_bin=new CBinaryLabels(Y_bin);
+			SG_REF(feat_train_bin);
+			SG_REF(feat_test_bin);
+			SG_REF(label_train_bin);
+
+			/* regression data */
+			SGMatrix<float64_t> X_reg(2, N_TRAIN);
+			SGMatrix<float64_t> X_test_reg(2, N_TEST);
+			SGVector<float64_t> Y_reg(N_TRAIN);
+
+			for (index_t i=0; i<N_TRAIN; ++i)
+			{
+				X_reg(0,i)=CMath::random(-1.0, 1.0);
+				X_reg(1,i)=CMath::random(-1.0, 1.0);
+				Y_reg[i]=3*X_reg(0,i)+4*X_reg(1,i);
+			}
+
+			for (index_t i=0; i<N_TEST; ++i)
+			{
+				X_test_reg(0,i)=CMath::random(-1.0, 1.0);
+				X_test_reg(1,i)=CMath::random(-1.0, 1.0);
+			}
+
+			feat_train_reg=new CDenseFeatures<float64_t>(X_reg);
+			feat_test_reg=new CDenseFeatures<float64_t>(X_test_reg);
+			label_train_reg=new CRegressionLabels(Y_reg);
+			SG_REF(feat_train_reg);
+			SG_REF(feat_test_reg);
+			SG_REF(label_train_reg);
+		}
+
+		static void TearDownTestCase() {
+			SG_UNREF(feat_train_bin);
+			SG_UNREF(feat_test_bin);
+			SG_UNREF(label_train_bin);
+
+			SG_UNREF(feat_train_reg);
+			SG_UNREF(feat_test_reg);
+			SG_UNREF(label_train_reg);
+		}
+
+		static const index_t N_TRAIN=30;
+		static const index_t N_TEST=10;
+
+		static CDenseFeatures<float64_t>* feat_train_bin;
```

Comment:
I think the feature pointers should be the same for regression and classification